jython/DispatcherSystem/MoveTrain.py

```diff
@@ -0,0 +1,598 @@
+                aat.setStopBySpeedProfileAdjust(info.getStopBySpeedProfileAdjust())
+                aat.setUseSpeedProfile(info.getUseSpeedProfile())
+                if (not aat.initialize()):
+                    log.error("ERROR initializing autorunning for train {}", at.getTrainName())
+                    JOptionPane.showMessageDialog(dispatcherFrame, jmri.jmrit.dispatcher.Bundle.getMessage(
+                            "Error27", at.getTrainName()), jmri.jmrit.dispatcher.Bundle.getMessage("MessageTitle"),
+                            JOptionPane.INFORMATION_MESSAGE)
+                    return -1
+                DF.getAutoTrainsFrame().addAutoActiveTrain(aat)
+            DF.allocateNewActiveTrain(at)
+            DF.newTrainDone(at)
+
+        else:
+            log.warn("failed to create Active Train '{}'", info.getTrainName())
+            return -1
+        return 0
+            
+    ## ***********************************************************************************
+    
+    ## sound routines
+    
+    ## ***********************************************************************************            
+        
+            
+    # use external "nircmd" command to "speak" some text  (I prefer this voice to eSpeak)
+    def speak(self,msg) :
+        #if self.logLevel > 0: print("about to speak",msg)
+        #java.lang.Runtime.getRuntime().exec('Z:\\ConfigProfiles\\jython\\sound2\\nircmd speak text "' + msg +'"')    
+        my_dir = jmri.util.FileUtil.getExternalFilename('program:jython/DispatcherSystem/programs')
+        if self.logLevel > 0: print "nircmd" + my_dir+'/nircmd'
+        java.lang.Runtime.getRuntime().exec(my_dir+'/nircmd speak text "' + msg +'"')
+        return
+        
+    def announce(self, fromblockname, toblockname, speak_on): 
+
+        from_station = self.get_station_name(fromblockname)
+        to_station = self.get_station_name(toblockname)
+
+        if speak_on == True:
+            self.speak("The train in "+ from_station + " is due to depart to " + to_station)
+            
+    def get_station_name(self, block_name):
+        BlockManager = jmri.InstanceManager.getDefault(jmri.BlockManager)
+        block = BlockManager.getBlock(block_name)
+        comment = block.getComment()
+        # % is the delimeter for block name
+        delimeter = '"'
+        if delimeter in comment:
+            station_name = self.get_substring_between_delimeters(comment, delimeter)
+        else:
+            station_name = block_name
+        return station_name
+        
+    def get_substring_between_delimeters(self, comment, delimeter):
+        start = delimeter
+        end = delimeter
+        s = comment
+        substring = s[s.find(start)+len(start):s.rfind(end)]
+        return substring
+        
+        
+    def bell(self, bell_on = "True"):
+        if bell_on == "True":
+            snd = jmri.jmrit.Sound("resources/sounds/Bell.wav")
+            snd.play()
+
+            
+    
```

Comment:
Spotbugs and Github complains about the last line doesn't end with a new line. The cause of this is that the last line has spaces after the new line. The last line should have only a new line and no spaces.

Reply:
This error is important to fix since CI will not pass until it's fixed.

Reply:
have modified

Reply:
Your edit did not solve the problem, but I have edited the file on GitHub. Your branch now has an extra commit from me on the server that you need to fetch.

Do you use command line "git" or "GitHub Desktop" or some other git tool?

Reply:
I used Netbeans to push the commit, which I find a pain, but netbeans is quite useful to generate a system that I can test with. I use github desktop to check that the edits go through, by opening up atom and having a look. I need a much better system.

Reply:
I use NetBeans for development but GitHub Desktop to make the commits.

I have read several comments about the git client within NetBeans doesn't work well with JMRI.

Here is a tutorial about using GitHub Desktop:
https://www.jmri.org/help/en/html/doc/Technical/githubdesktopinto.shtml